### PR TITLE
LF : Fix/Clean some LF related TODOs

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: navigator/backend/src/main/scala/com/digitalasset/navigator/query/project/package.scala [hunk 1]
```diff
@@ -48,40 +48,46 @@ object project {
...
-          }
-          next match {
-            case Right((nextType, nextCursor)) => loop(nextType, nextCursor, ps)
-            case Left(e) => Left(e)
+          val tyCon = tc.name.identifier
+          ps(tyCon) match {
+            case Some(ddt) =>
+              val next = tc.instantiate(ddt) match {
+                case DamlLfEnum(_) =>
+                  Right(StringValue(tyCon.toString))
```
Comment (stefanobaghino-da) [inline navigator/backend/src/main/scala/com/digitalasset/navigator/query/project/package.scala:67]: Shouldn't we add a test for this?

**Review by stefanobaghino-da — COMMENTED** [review]
Had a look at the navigator part, should we add a test for that?

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionCoderSpec.scala [hunk 2]
```diff
@@ -39,7 +39,7 @@ class TransactionCoderSpec
 
   "encode-decode" should {
 
-    "do contractInstance" in {
+    "do contractInstance" ignore {
```
Comment (cocreature) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionCoderSpec.scala:42]: Why is encoding/decoding flaky? That sounds very suspicious

Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionCoderSpec.scala]: reverted

File: navigator/backend/src/main/scala/com/digitalasset/navigator/query/project/package.scala (hunk 1 above)
Comment (rautenrieth-da) [reply navigator/backend/src/main/scala/com/digitalasset/navigator/query/project/package.scala:67]: Ideally, yes. The code from this file is tested by [ContractSortSpec.scala](https://github.com/digital-asset/daml/blob/main/navigator/backend/src/test/scala/com/digitalasset/navigator/backend/query/ContractSortSpec.scala), but the test coverage is very bad. In particular, it does not cover the `checkParameter` method that was modified by this PR.

BTW, the entire custom query engine should (in my opinion) be replaced by SQL queries, so I'm not sure how much we should invest in improving the testing of the engine.

Comment (rautenrieth-da) [reply navigator/backend/src/main/scala/com/digitalasset/navigator/query/project/package.scala:67]: Adding a test:

You can extend [ContractSortSpec.scala](https://github.com/digital-asset/daml/blob/main/navigator/backend/src/test/scala/com/digitalasset/navigator/backend/query/ContractSortSpec.scala) with the following lines at the bottom:

```
  test(List("template.parameter.foo" -> ASCENDING), List(contract2, contract4, contract1, contract3))
  test(List("template.parameter.int" -> ASCENDING), List(contract1, contract3, contract2, contract4))
```

The first line sorts the test contracts by `template.parameter.foo`, i.e., by the `foo` field of the contract parameter. That field has value `text` for the contracts 1 and 3, and does not exist for contracts 2 and 4.

You could then extend the test coverage by changing [damlLfRecord0](https://github.com/digital-asset/daml/blob/main/navigator/backend/src/test/scala/com/digitalasset/navigator/backend/query/ContractSortSpec.scala#L32) and [damlLfRecord1](https://github.com/digital-asset/daml/blob/main/navigator/backend/src/test/scala/com/digitalasset/navigator/backend/query/ContractSortSpec.scala#L39) to include other data types, and adding additional test cases for them.

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionCoderSpec.scala [hunk 3]
```diff
@@ -383,37 +383,37 @@ class TransactionCoderSpec
...
-        case ((nodeId, node), version1, version2) =>
-          whenever(version1 != version2) {
-            val (txVersion, nodeVersion) = inIncreasingOrder(version1, version2)
+      val postV10versions = TransactionVersion.All.filter(_ > V10)
 
-            val normalizedNode = minimalistNode(nodeVersion)(node).updateVersion(nodeVersion)
+      def twoDifferentVersions =
+        for {
+          v1 <- Gen.oneOf(postV10versions)
+          v2 <- Gen.oneOf(postV10versions.filter(_ != v1))
```
Comment (hurryabit) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionCoderSpec.scala:393]: How about
```suggestion
          v2 <- Gen.oneOf(postV10versions.filter(_ > v1))
```
and remove the call to `inIncreasingOrder` below? This seems to express the intent more clearly.

File: extractor/BUILD.bazel [hunk 4]
```diff
@@ -27,9 +28,7 @@ TEST_DARS = [
     daml_compile(
         name = darmod,
         srcs = ["src/test/resources/damls/%s.daml" % darmod],
-        # FixMe: https://github.com/digital-asset/daml/issues/2289
-        # change version to lf_latest_version when freezing numeric in the next language version
-        target = lf_dev_version,
+        target = lf_latest_version,
```
Comment (hurryabit) [inline extractor/BUILD.bazel:31]: `Numeric` is already in DAML-LF 1.8, isn't it? So, I guess we just missed this.

**Review by hurryabit — COMMENTED** [review]
Except for the navigator test, which I can't really judge, this looks good to me. If @stefanobaghino-da and @rautenrieth-da are happy with the changes there, let's merge. Thank you.

**Review by hurryabit — APPROVED** [review]

File: extractor/BUILD.bazel (hunk 4 above)
Comment (remyhaemmerle-da) [reply extractor/BUILD.bazel:31]: Yes.

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionCoderSpec.scala (hunk 3 above)
Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/TransactionCoderSpec.scala:393]: Good idea.

**Review by rautenrieth-da — APPROVED** [review]
Thanks for taking the time to add the Navigator tests!

**Pull request merged**